File: Visual Studio/C#/Eruru.ChatRobotRPC For Shared Project/ChatRobotMessage.cs
using System;

namespace Eruru.ChatRobotAPI {

	public class ChatRobotMessage {

		public ChatRobotMessageType Type { get; set; }
		public long Robot { get; set; }
		public long QQ { get; set; }
		public long Group { get; set; }
		public string Text { get; set; }
		public long Number { get; set; }
		public long ID { get; set; }
		public DateTime DateTime { get; }

		public ChatRobotMessage (ChatRobotMessageType type, long robot, long group, long qq, string text, long number, long iD) {
			Type = type;
			Robot = robot;
			QQ = qq;
			Group = group;
			Text = text;
			Number = number;
			ID = iD;
			DateTime = DateTime.Now;
		}

		public void Reply (string message) {
			Reply (message, ChatRobotSendMessageType.Text);
		}
		public void Reply (object message) {
			Reply (message?.ToString ());
		}

		public void ReplyJson (string message) {
			Reply (message, ChatRobotSendMessageType.Json);
		}
		public void ReplyJson (object message) {
			ReplyJson (message?.ToString ());
		}

		public void ReplyXml (string message) {
			Reply (message, ChatRobotSendMessageType.Xml);
		}
		public void ReplyXml (object message) {
			ReplyXml (message?.ToString ());
		}

		public static void Send (ChatRobotMessageType type, long robot, string message, long group = default, long qq = default, ChatRobotSendMessageType sendType = ChatRobotSendMessageType.Text) {
			switch (type) {
				case ChatRobotMessageType.Friend:
					switch (sendType) {
						case ChatRobotSendMessageType.Text:
							ChatRobotAPI.SendFriendMessage (robot, qq, message);
							break;
						case ChatRobotSendMessageType.Json:
							ChatRobotAPI.SendFriendJsonMessage (robot, qq, message);
							break;
						case ChatRobotSendMessageType.Xml:
							ChatRobotAPI.SendFriendXmlMessage (robot, qq, message);
							break;
						default:
							throw new NotImplementedException (sendType.ToString ());
					}
					break;
				case ChatRobotMessageType.Group:
					switch (sendType) {
						case ChatRobotSendMessageType.Text:
							ChatRobotAPI.SendGroupMessage (robot, group, message);
							break;
						case ChatRobotSendMessageType.Json:
							ChatRobotAPI.SendGroupJsonMessage (robot, group, message);
							break;
						case ChatRobotSendMessageType.Xml:
							ChatRobotAPI.SendGroupXmlMessage (robot, group, message);
							break;
						default:
							throw new NotImplementedException (sendType.ToString ());
					}
					break;
				case ChatRobotMessageType.Discuss:
					switch (sendType) {
						case ChatRobotSendMessageType.Text:
							ChatRobotAPI.SendDiscussMessage (robot, group, message);
							break;
						case ChatRobotSendMessageType.Json:
							ChatRobotAPI.SendDiscussJsonMessage (robot, group, message);
							break;
						case ChatRobotSendMessageType.Xml:
							ChatRobotAPI.SendDiscussXmlMessage (robot, group, message);
							break;
						default:
							throw new NotImplementedException (sendType.ToString ());
					}
					break;
				default:
					throw new NotImplementedException (type.ToString ());
			}
		}
		public static void Send (ChatRobotMessageType type, long robot, object message, long group = default, long qq = default, ChatRobotSendMessageType sendType = ChatRobotSendMessageType.Text) {
			Send (type, robot, message?.ToString (), group, qq, sendType);
		}

		void Reply (string message, ChatRobotSendMessageType type) {
			Send (Type, Robot, message, Group, QQ, type);
		}

	}

}

File: Intelij IDEA/Java/Eruru.ChatRobotRPC/src/main/java/org/eruru/chatrobotrpc/ChatRobotMessage.java
package org.eruru.chatrobotrpc;

import java.util.Date;

public class ChatRobotMessage {

	private ChatRobotMessageType type;
	private long robotQQ;
	private long qq;
	private long group;
	private String text;
	private long number;
	private long id;
	private Date date;

	public ChatRobotMessage (ChatRobotMessageType type, long robotQQ, String text, long number, long id, long qq, long group) {
		this.setType (type);
		this.setRobotQQ (robotQQ);
		this.setQQ (qq);
		this.setGroup (group);
		this.setText (text);
		this.setNumber (number);
		this.setId (id);
		this.setDate (new Date ());
	}

	public ChatRobotMessageType getType () {
		return type;
	}

	public void setType (ChatRobotMessageType type) {
		this.type = type;
	}

	public long getRobotQQ () {
		return robotQQ;
	}

	public void setRobotQQ (long robotQQ) {
		this.robotQQ = robotQQ;
	}

	public long getQQ () {
		return qq;
	}

	public void setQQ (long qq) {
		this.qq = qq;
	}

	public long getGroup () {
		return group;
	}

	public void setGroup (long group) {
		this.group = group;
	}

	public String getText () {
		return text;
	}

	public void setText (String text) {
		this.text = text;
	}

	public long getNumber () {
		return number;
	}

	public void setNumber (long number) {
		this.number = number;
	}

	public long getId () {
		return id;
	}

	public void setId (long id) {
		this.id = id;
	}

	public Date getDate () {
		return date;
	}

	public void setDate (Date date) {
		this.date = date;
	}

	public void Reply (String message) {
		reply (message, false);
	}

	public void replyJson (String message) {
		reply (message, true);
	}

	private void reply (String message, boolean isJson) {
		switch (getType ()) {
			case Friend:
				if (isJson) {
					ChatRobotAPI.sendFriendJsonMessage (robotQQ, qq, message);
					break;
				}
				ChatRobotAPI.sendFriendMessage (robotQQ, qq, message);
				break;
			case Group:
				if (isJson) {
					ChatRobotAPI.sendGroupJsonMessage (robotQQ, group, message);
					break;
				}
				ChatRobotAPI.sendGroupMessage (robotQQ, group, message);
				break;
		}
	}

}
